client/js/app.js

```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
     }
 }
 
-window.onload = function () {
+function getColumnDataSuccess(response) {
+    let data = response.columns;
+    let keys = Object.keys(data);
+    let table = document.getElementById('column_body');
+    while (table.firstChild) {
+        table.removeChild(table.firstChild);
+    }
+    keys.forEach(key => {
+        let tr = document.createElement("tr");
+        let td1 = document.createElement("td");
+        td1.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td1.append(key.replace(/_/g, " "));
+        tr.appendChild(td1);
+        let td2 = document.createElement("td");
+        td2.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td2.append(data[key].type);
+        tr.appendChild(td2);
+        table.appendChild(tr);
+    });
+}
+
+function getColumnData(dataset) {
+    loadAdditionalData("columns", "POST", {dataset: dataset}, getColumnDataSuccess, () => {
+        // TODO handle error
+        setLoading(false)
+    });
+}
+
+function onExampleSelected(example) {
+    console.log(example.textContent);
+    let ul = document.getElementById("dataset-list");
+    for (var i = 0; i < ul.children.length; i++) {
+        if (ul.children[i].dataset.selected === 'true') {
+            var dataset = ul.children[i].dataset.val;
+            break;
+        }
+    }
```

Comment:
could you extract this to a function `getCurrentDataset()` and use this function in all places where a `dataset` is needed?